### PR TITLE
[MRG] Rename library to sourmash

### LINKED ISSUE #162: Renaming sourmash_lib module to sourmash
The `sourmash_lib` name seems to come from the project early days, when the entry point was the [`sourmash`][1] 'executable' in the project root (for dev purposes).

If we use `python setup.py develop` we don't need the 'executable', and then we can use the name instead of `sourmash_lib`

[1]: https://github.com/dib-lab/sourmash/blob/564f89d319e899fe5220c9644d7984e1b1f9053f/sourmash

### PR DESCRIPTION
**Pull request by luizirber**
Change library name to sourmash instead of sourmash_lib
  * sourmash_lib is still available for older scripts
  * Remove the local sourmash script
  * Fix tests that expected the local sourmash script

Supersedes #164 
Fixes #162 

- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by luizirber** [comment]
Ready for review @ctb

**Comment by luizirber** [comment]
This turned out to be easier than expected. The important lesson is avoiding doing absolute imports inside the module, so instead of doing
```python
from sourmash import foo
```
do
```python
from . import foo
```
instead. For external/testing code, continue doing either
```python
from sourmash import foo
```
or 
```python
from sourmash_lib import foo
```

File: setup.py [hunk 1]
```diff
@@ -51,16 +51,16 @@
     "author": "C. Titus Brown",
     "author_email": "titus@idyll.org",
     "license": "BSD 3-clause",
-    "packages": ["sourmash_lib"],
+    "packages": find_packages(),
```
Comment (standage) [inline setup.py]: Presumably recursively finds all packages and subpackages?

File: sourmash [hunk 2]
```diff
@@ -1,5 +0,0 @@
-#! /usr/bin/env python
-# NOTE: this script is only used in 'developer' mode (python setup.py
-# build_ext -i) -- see setup.py entrypoints for what end-users use.
-from sourmash_lib.__main__ import main
-main()
```
Comment (standage) [inline sourmash]: Is there no command line entry point now? Or should we do something like `pip install -e .` to set up a dev environment?

**Comment by ctb** [comment]
On Fri, Mar 09, 2018 at 09:37:36PM -0800, Daniel Standage wrote:
> > @@ -1,5 +0,0 @@
> -#! /usr/bin/env python
> -# NOTE: this script is only used in 'developer' mode (python setup.py
> -# build_ext -i) -- see setup.py entrypoints for what end-users use.
> -from sourmash_lib.__main__ import main
> -main()
> 
> Is there no command line entry point now? Or should we do something like `pip install -e .` to set up a dev environment?

yep!

**Comment by ctb** [comment]
great!

can we/should we put in a deprecation warning for 'import sourmash_lib' or
wait until later releases?

**Comment by ctb** [comment]
Note: if developers get an error `AttributeError: 'SourmashSignature' object has no attribute 'get'`, then they need to do a `pip install -e .` again to update the install to point at `sourmash` instead of `sourmash_lib`.

File: setup.py (hunk 1 above)
Comment (luizirber) [reply setup.py]: Yes, it is recommended by the Python packaging guide: https://packaging.python.org/tutorials/distributing-packages/#packages

File: sourmash (hunk 2 above)
Comment (luizirber) [reply sourmash]: You need `pip install -e .` (or `python setup.py develop`) to have the `sourmash` command available (it uses the entry point defined in `setup.py`).

**Comment by luizirber** [comment]
@ctb yes, it is better to 'reset' your environment after this changes (reinstall sourmash with `pip install -e .`). I had some really weird test errors while developing because the environment still had parts from the old install...

**Comment by ctb** [comment]
yes, exactly.

**Pull request merged**